### PR TITLE
Upgrade to Go 1.12 and switch to modules

### PR DESCRIPTION
**Pull request by unguiculus**

### CODE REVIEW

**Comment by scottrigby** [comment]
Speaking of Circle, we still need to test PRs. I added #30 to track this

**Comment by rimusz** [comment]
these two lines https://github.com/helm/chart-releaser/blob/master/.circleci/config.yml#L10-L11 should be replaced with:
```
- run: go mod download
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,11 +4,10 @@ jobs:
     # `dep ensure` must be run from within a known GOPATH/src.
     working_directory: /go/src/github.com/helm/chart-releaser
```
Comment (scottrigby) [inline .circleci/config.yml]: Nice. The working directory doesn't matter now with go modules, right?

Comment (unguiculus) [reply .circleci/config.yml]: Yes, but speaking of with, we should remove it so we are not inside GOPATH. Otherwise, we'd have to set GO111MODULE=off.

**Pull request merged**